### PR TITLE
Ch toptica functionality

### PR DESCRIPTION
**Pull request by CatherineH**
This PR contains several updates relating to toptica topmode charm controller.
this includes:
- fixing a command spelling error
- adding the udev rule
- adding firmware and topmode charm controller serial number
- handling the situation when no laser is plugged into the controller

### CODE REVIEW

**Comment by CatherineH** [comment]
The firmware map() does not pass pylinting, I'm not sure how his work in cc1

**Comment by scasagrande** [comment]
What `map()` are you referring too?

**Comment by CatherineH** [comment]
On the firmware parsing.

**Comment by scasagrande** [comment]
Ah, I see in the commit history now. You have to go `from builtins import map` at the top. This is because `map` changed between Py2 and Py3 to return an iterator instead of a list. So to make the results consistent between the two, we import the new implementation.

File: doc/examples/toptica/ex_topticatopmode.py [hunk 1]
```diff
@@ -10,8 +10,10 @@
 if system() == 'Windows':
       tm = ik.toptica.TopMode.open_serial('COM14', 115200)
 else:
-      tm = ik.toptica.TopMode.open_serial('/dev/ttyACM0', 115200)
+      tm = ik.toptica.TopMode.open_serial('/dev/toptica_laser', 115200)
```
Comment (scasagrande) [inline doc/examples/toptica/ex_topticatopmode.py]: Maybe include a comment referencing the udev rules

File: instruments/toptica/topmode.py [hunk 2]
```diff
@@ -137,11 +139,23 @@ def enable(self):
         @enable.setter
         def enable(self, newval):
             if not isinstance(newval, bool):
-                raise TypeError(
-                    "Laser emmission must be a boolean, got: {}".format(newval))
+                raise TypeError("Emission status must be a boolean, got: {}".
```
Comment (scasagrande) [inline instruments/toptica/topmode.py]: Let's not line break on the `.`, instead do it just before the word "got" for code readability

File: instruments/toptica/topmode.py [hunk 3]
```diff
@@ -137,11 +139,23 @@ def enable(self):
...
+                                format(type(newval)))
+            if not self.is_connected:
+                return
             self.parent.set(self.name + ":enable-emission", newval)
 
         @property
+        def is_connected(self):
+            """
+            Check whether a laser is even connected
```
Comment (scasagrande) [inline instruments/toptica/topmode.py]: :type: `bool`

File: instruments/toptica/topmode.py [hunk 4]
```diff
@@ -137,11 +139,23 @@ def enable(self):
...
+            """
+            if self.serial_number == 'unknown':
+                raise RuntimeError("Laser was not recognized by charm "
+                                   "controller. Is it plugged in?")
+            return True
+
+        @property
         def on_time(self):
             """
             Gets the 'on time' value for the laser
```
Comment (scasagrande) [inline instruments/toptica/topmode.py]: Doc string confuses me, it makes me thing I'm getting back a value for how long the laser has been on for or will be on for.

**Comment by scasagrande** [comment]
Looks good, just a few minor things. I want to get all the usbtmc bugfixes into master first, then we'll repoint this PR at `develop` branch and try to include a few things in together for a `v0.2.0` release.

**Comment by scasagrande** [comment]
I'm going to close and reopen this PR to merge in develop instead of master

**Pull request closed without merging**